sycl/doc/extensions/SPIRV/SPV_INTEL_spec_conditional.asciidoc

```diff
@@ -0,0 +1,694 @@
+    *OpBranchSpecConstantWithElseINTEL* instruction.
+
+  - Otherwise, verify that _Condition i_ has the value zero.
+
+  - If _Variable i_ is defined by a block in the "then" range of
+    *OpBranchSpecConstantINTEL* or *OpBranchSpecConstantWithElseINTEL*, or if
+    _Variable i_ is defined by a block in the "else" range of
+    *OpBranchSpecConstantWithElseINTEL*, verify that _Parent i_ is contained by
+    that same "then" or "else" range.
+
+* For each specialization constant (*OpSpecConstantXXX*) that is decorated with
+  *ConditionalINTEL*, verify that the specialization constant's _<id>_ is not
+  also used as a _Condition_ for *OpConditionalCapabilityINTEL*,
+  *OpBranchSpecConstantINTEL*, *OpBranchSpecConstantWithElseINTEL*,
+  *OpPhiSpecConstantINTEL*, or *ConditionalINTEL*.
+
+*TODO*: I'm looking for feedback on the validation rules that involve
+comparison of abstract specialization constants such as "validate that `G(i)`
+can never be true unless `G(c)` is also true".  These validations are likely
+very difficult in the general case, but most cases will be very easy.  (In most
+cases, I think that `G(i)` and `G(c)` will both be so simple that it will be
+easy to prove whether one implies the other.)  How should the spec handle this?
+Are the validation rules _requirements_ that a validation tool must perform, or
+are they just rules about what SPIR-V is not valid?  If they are not
+requirements, then we could leave the wording as I have it and then each
+validation tool would be free to implement the validation checks as thoroughly
+as it wants.
+
+
+== New tokens defined by this extension
+
+=== New capabilities
+
+The module must declare that it uses the following capability in order to use
+any of the decorations or instructions defined in the sections below.  If
+declared, this capability must be unconditionally declared via *OpCapability*.
+It may not be conditionally declared via *OpConditionalCapabilityINTEL*.
+
+[cols="1,15,5",options="header",width="100%"]
+|===
+2+^| Capability | Implicitly Declares
+| ????
+| *SpecConditionalINTEL* +
+Module conditionally enables code based on the value of a specialization
+constant.
+|
+|===
+
+=== New decorations
+
+[cols="1,10,5,5",options="header",width="100%"]
+|====
+2+^| Decoration | Extra Operands | Enabling Capabilities
+| ????
+| *ConditionalINTEL* +
+May be applied only to *OpFunction*, global (module scope) *OpVariable*, type
+declarations (*OpTypeXXX*), or constant instructions (*OpConstantXXX* or
+*OpSpecConstantXXX*).  Indicates that the decorated instruction must be removed
+if the value of the specialization constant identified by _Condition_ is
+*false*.  The _Condition_ must be a _Boolean type_ scalar.
+
+If the decorated instruction is *OpFunction*, the function and all of the
+instructions it contains are removed when the specialization constant is
+*false*.  If the function has an associated *OpEntryPoint*, that is also
```

Comment:
My understanding is that this section is the most formal part of the spec, so it should probably contain other details, like removing of associated `OpName` and `OpMemberName` instructions. BTW, what about debug info?

Reply:
Resolving this because we decided to go with an alternate design that does not require a SPIR-V extension.